SalesforceCommerceCloud/pwa-kit
# Use Prompted SiteId in Generated Configuration Files

# Description

Currently when generating a project it's possible to have a generated project referencing incorrect business manager sites. For example, if you generate a new project without using any kind of answer presets, you'll end up with `config/default.js` and `config/sites.js` referencing `RefArch`.

In this PR I ensure that the `defaultSite` in the `default.js` file and the site `id` value located in the `sites.js` file are templated. This results in the generated files having the correct values. 

_NOTE: This however doesn't fix any issues resulting from the actual configuration of the site being different than what is in Business Manger. That will be handled in a future ticket where we retrieve the site information from business manager itself._ 

# Types of Changes




# Changes

- Template `defaultSite` in the `config/defualt.js` file.
- Template the `id` value in the `/config/sites.js` file.
- Removed unused `api.config.json` file

# How to Test-Drive This PR

Part 1.
```
1. Generate a project using a preset like this `GENERATOR_PRESET=test-project packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js --outputDir /tmp/generated-project`
2. Verify that `defaultSite` in the defualt.js config file is set to `RefArch`
3. Also verify that the `id` value for the only site in the `sites.js` file is also `RefArch`
```

Part 2.
```
1. Generate a project NOT using a preset like this `packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js --outputDir /tmp/generated-project`
2. Answer all the questions and use a unique value for the site id when asked.
3. Verify that `defaultSite` in the defualt.js config file is set to the unique value above.
4. Also verify that the `id` value for the only site in the `sites.js` file is also the unique value.
```

# Checklists




## General
